src/GoogleCloudPubSub/GoogleCloudPubSub.ts

```diff
@@ -199,6 +199,23 @@ export class GoogleCloudPubSub implements GCPubSub {
     });
   }
 
+  /**
+   * Stop listening to a specific subscription. Close the server connection.
+   * @param event Event name
+   */
+  public async unsubscribe(event: string): Promise<void> {
+    const subscriptionName: string = this.getSubscriptionName(event);
+    // Cover a case where there could be a custom subscription name with a prefix.
+    const cachedSubscription: Subscription | undefined =
+      this.subscriptions.get(subscriptionName) || this.subscriptions.get(event);
```

Comment:
I think there is no need for the fallback since `getSubscriptionName` function returns `event` when the prefix is undefined
https://github.com/algoan/pubsub/blob/master/src/GoogleCloudPubSub/GoogleCloudPubSub.ts#L303

Reply:
Actually, if you use the `subscriptionOptions.name` option, you can choose a custom name for the subscription. So this cover the case where you choose to disable `subscriptionPrefix` and choose a custom name.

But I agree, this is twisted... this is part of the huge refactoring I want to make!